pkg/libinit/root_linux.go

```diff
@@ -263,24 +322,84 @@ func InstallModules(pattern string, exclude map[string]bool) error {
 	for _, filename := range files {
 		f, err := os.Open(filename)
 		if err != nil {
-			log.Printf("installModules: can't open %q: %v", filename, err)
+			log.Printf("InstallModules: can't open %q: %v", filename, err)
 			continue
 		}
-		// Module flags are passed to the command line in the form modulename.flag=val
+		// Module flags are passed to the command line in the from modulename.flag=val
 		// And must be passed to FileInit as flag=val to be installed properly
 		moduleName := strings.TrimSuffix(filepath.Base(filename), filepath.Ext(filename))
-		if _, ok := exclude[moduleName]; ok {
-			log.Printf("Skipping module %s", moduleName)
+		if loader.IsExcluded(moduleName) {
+			log.Printf("Skipping module %q", moduleName)
 			continue
 		}
 
 		flags := cmdline.FlagsForModule(moduleName)
-		err = kmodule.FileInit(f, flags, 0)
+		if err = kmodule.FileInit(f, flags, 0); err != nil {
+			log.Printf("InstallModules: can't install %q: %v", filename, err)
+		}
 		f.Close()
+	}
+
+	return nil
+}
+
+func readModules(f *os.File) []string {
+	r := bufio.NewReader(f)
+	modules := []string{}
+	for {
+		if i, err := r.ReadString('\n'); err == nil {
+			i = strings.TrimSpace(i)
+			if i == "" {
+				continue
+			} else if strings.HasPrefix(i, "#") {
+				continue
+			} else {
+				modules = append(modules, strings.TrimSuffix(i, "\n"))
+			}
+		} else if err != io.EOF {
+			log.Println("error on reading:", err)
+		} else {
```

Comment:
(https://pkg.go.dev/strings#TrimSpace) hmm so we need this because `ReadString returns err != nil if and only if the returned data does not end in delim` and the docs for TrimSpace state  `For simple uses, a Scanner may be more convenient.`

Perhaps it'll be clearer to restructure this code block as:

```
scanner := bufio.NewScanner(f)
for scanner.Scan() { // By default this uses newline delimiters
  // Do something with scanner.Text()
}
if err := scanner.Err(); err != nil {
  // Do something with err
}
```